Eclipse Ditto explorer UI

thfries opened the pull request:
Hello,
please find this draft pr for a ui for ditto. It is based on plain vanilla javascript and Bootstrap 5.
Happy to receive your feedback.

thjaeckle commented on ui/index.html:
```diff
@@ -0,0 +1,65 @@
+<html lang="en">
+<head>
+    <title>ditto explorer</title>
```
The title should be `Eclipse Ditto™ explorer`

thjaeckle commented on ui/index.html:
```diff
@@ -0,0 +1,65 @@
+<html lang="en">
```
Please add a copyright header to this file.

thjaeckle commented on ui/index.css:
```diff
@@ -0,0 +1,107 @@
+.table {
```
Please add a copyright header to this file.

thjaeckle commented on ui/modules/environments/authorization.html:
```diff
@@ -0,0 +1,58 @@
+<div class="modal fade bd-example-modal-sm" tabindex="-1">
```
Please add a copyright header to this file.

thjaeckle commented on ui/modules/environments/environments.html:
```diff
@@ -0,0 +1,7 @@
+<h5>Settings</h5>
```
Please add a copyright header to this file.

thjaeckle commented on ui/modules/things/features.html:
```diff
@@ -0,0 +1,73 @@
+<h5 data-bs-toggle="collapse" data-bs-target="#collapseFeatures">
```
Please add a copyright header to this file.

thjaeckle commented on ui/modules/things/fields.html:
```diff
@@ -0,0 +1,25 @@
+<div class="tab-pane container no-margin">
```
Please add a copyright header to this file.

thjaeckle commented on ui/modules/things/searchFilter.html:
```diff
@@ -0,0 +1,5 @@
+<div class="tab-pane container no-margin">
```
Please add a copyright header to this file.

thjaeckle commented on ui/modules/things/things.html:
```diff
@@ -0,0 +1,101 @@
+<div>
```
Please add a copyright header to this file.

thjaeckle commented on ui/modules/environments/environments.js:
```diff
@@ -0,0 +1,131 @@
+  eclipse_sandbox: {
+    api_uri: 'https://ditto.eclipseprojects.io',
+    bearer: '',
+    usernamePassword: null,
```
we could also pre-fill `'ditto:ditto'` as usernamePassword here ..

thfries replied:
I adapted that. Btw. I was not able to test with the sandbox, because of CORS restrictions. May be we should fully remove the second environment until this is resolved? Or we could set a sample for IoT Things?

thjaeckle replied:
Oh, agreed - remove the sandbox environment then.
The sandbox is currently not in a stable state anyways..

thfries replied:
ok, done. Thanks @thjaeckle

thjaeckle reviewed (changes requested):
Thanks for this great contribution @thfries 
I added a first review, mainly for missing copyright headers.

Apart from that we would happily accept this PR as a start for the UI 👍

thjaeckle reviewed (approved):
Looks good to me now, thanks for the prompt fixes 👍

thfries commented:
Thanks a lot for the improvements and feedback. Really appreciate that.
Fully agree to all points.
- The headline would include defining a short label for each field, correct?
- A new thing with custom thingId can be created on the "Thing JSON" tab. But that is not obvious. I also thought about another option to create the sample thing instead of an empty thing (the same like in the swagger UI).

thfries commented:
Hi @thjaeckle,
I was active last weekend and implemented almost all you proposed changes. The left area looks a bit crowded now but the the right part contains only aspects of the thing:
<img width="1435" alt="table_headers" src="https://user-images.githubusercontent.com/20593339/175864307-9afd059b-959b-4bc6-9f08-c1f4d9d457b3.png">
The missing part is the thing creation with a given id.
I added a label for the field. There are many more future enhancements for the fields possible, like easily generate filters for the search, sorting or custom formatting.
Please let me know how you would like to proceed with the PR. As there are several new changes in the commit, I would prefer to create a new branch for that.
Thank you.

thjaeckle commented:
@thfries that looks great 👍 

I propose that we merge this already accepted PR and you create another PR for the improvements.
Thank you so much for working and contributing the UI.
